distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/SystemStateManager.java

```diff
@@ -0,0 +1,208 @@
+                serviceManager.startFeature(false, feature);
+            }
+            serviceManager.waitForAllBundles();
+            //we try a couple of time here because some features might not stop the first time
+            //due to dependencies
+            List<String> stoppedFeatures = new ArrayList<>();
+            for (int i = 0; i < FEATURE_STOP_RETRY_COUNT; i++) {
+                stoppedFeatures.clear();
+                for (String feature : featuresToStop) {
+                    LOGGER.debug("Stopping feature {}", feature);
+                    try {
+                        serviceManager.stopFeature(false, feature);
+                        stoppedFeatures.add(feature);
+                    } catch (Exception e) {
+                        LOGGER.debug("Failed to stop feature {}", feature);
+                    }
+                }
+                featuresToStop.removeAll(stoppedFeatures);
+            }
+            serviceManager.waitForAllBundles();
+
+            //reset the configurations
+            Configuration[] configs = adminConfig.listConfigurations(CONFIGURATION_FILTER);
+            Map<String, Configuration> currentConfigurations = new HashMap<>();
+            for (Configuration config : configs) {
+                currentConfigurations.put(config.getPid(), config);
+            }
+            Map<String, Configuration> addedConfigs = new HashMap<>(currentConfigurations);
+            addedConfigs.keySet()
+                    .removeAll(baseConfigurations.keySet());
+            for (Configuration config : addedConfigs.values()) {
+                LOGGER.debug("Deleting configuration {}", config.getPid());
+                config.delete();
+            }
+
+            Map<String, Configuration> removedConfigs = new HashMap<>(baseConfigurations);
+            removedConfigs.keySet()
+                    .removeAll(currentConfigurations.keySet());
+            for (Configuration config : removedConfigs.values()) {
+                LOGGER.debug("Adding configuration {}", config.getPid());
+                Configuration newConfig = adminConfig.getConfiguration(config.getPid(), null);
+                newConfig.update(config.getProperties());
+            }
+
+            for (Configuration config : baseConfigurations.values()) {
+                if (currentConfigurations.containsKey(config.getPid())
+                        && !propertiesMatch(config.getProperties(),
+                        currentConfigurations.get(config.getPid())
+                                .getProperties())) {
+                    LOGGER.debug("Updating configuration {}", config.getPid());
+                    config.update(baseConfigurations.get(config.getPid())
+                            .getProperties());
+                }
+            }
+
+            serviceManager.waitForAllBundles();
+
+            //reset the catalog
+            console.runCommand("catalog:removeall -f -p");
```

Comment:
If we hardcode wiping the catalog like this, it's probably worth a javadoc, since the API doesn't make it obvious that it wipes the catalog.

I'm also wondering if we should reset the catalog instead of wiping it. Could do a `catalog:export -p` in `captureSystemState()` and a `catalog:ingest -p` after the wipe. That would make this a little more generic and reusable. 

Does the registry regenerate its metacard for the local node automatically now, or will this still put it in a weird state if it's part of the captured system?